[2025-09-20T18:48:26.902+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-09-20T18:48:26.943+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: bronze_full_load_spaceparts_optimized.extract_load_dim_Order_DocType scheduled__2024-01-01T00:00:00+00:00 [queued]>
[2025-09-20T18:48:26.960+0000] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: bronze_full_load_spaceparts_optimized.extract_load_dim_Order_DocType scheduled__2024-01-01T00:00:00+00:00 [queued]>
[2025-09-20T18:48:26.963+0000] {taskinstance.py:2867} INFO - Starting attempt 1 of 3
[2025-09-20T18:48:27.002+0000] {taskinstance.py:2890} INFO - Executing <Task(PythonOperator): extract_load_dim_Order_DocType> on 2024-01-01 00:00:00+00:00
[2025-09-20T18:48:27.015+0000] {standard_task_runner.py:72} INFO - Started process 156 to run task
[2025-09-20T18:48:27.046+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'bronze_full_load_spaceparts_optimized', 'extract_load_dim_Order_DocType', 'scheduled__2024-01-01T00:00:00+00:00', '--job-id', '134', '--raw', '--subdir', 'DAGS_FOLDER/01_Bronce.py', '--cfg-path', '/tmp/tmp_0oj3zm9']
[2025-09-20T18:48:27.050+0000] {standard_task_runner.py:105} INFO - Job 134: Subtask extract_load_dim_Order_DocType
[2025-09-20T18:48:27.465+0000] {task_command.py:467} INFO - Running <TaskInstance: bronze_full_load_spaceparts_optimized.extract_load_dim_Order_DocType scheduled__2024-01-01T00:00:00+00:00 [running]> on host 6c581f43b95a
[2025-09-20T18:48:27.586+0000] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='data-team' AIRFLOW_CTX_DAG_ID='bronze_full_load_spaceparts_optimized' AIRFLOW_CTX_TASK_ID='extract_load_dim_Order_DocType' AIRFLOW_CTX_EXECUTION_DATE='2024-01-01T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-01-01T00:00:00+00:00'
[2025-09-20T18:48:27.587+0000] {taskinstance.py:732} INFO - ::endgroup::
[2025-09-20T18:48:27.603+0000] {01_Bronce.py:381} INFO - Starting optimized extraction for table: dim.Order-DocType
[2025-09-20T18:48:30.755+0000] {01_Bronce.py:394} INFO - Table dim.Order-DocType has 4 rows
[2025-09-20T18:48:30.756+0000] {01_Bronce.py:416} INFO - Using chunk size: 50,000 for 4 rows
[2025-09-20T18:48:30.756+0000] {01_Bronce.py:421} INFO - Starting streaming extraction and load for dim.Order-DocType
[2025-09-20T18:48:33.029+0000] {01_Bronce.py:231} INFO - Processing chunk 1 with 4 rows
[2025-09-20T18:48:33.030+0000] {01_Bronce.py:242} ERROR - Error extracting data from dim.Order-DocType: module 'pandas' has no attribute 'np'
[2025-09-20T18:48:33.031+0000] {01_Bronce.py:356} ERROR - Error in streaming load to bronze_dim_order_doctype: module 'pandas' has no attribute 'np'
[2025-09-20T18:48:33.031+0000] {01_Bronce.py:441} ERROR - Error processing table dim.Order-DocType: module 'pandas' has no attribute 'np'
[2025-09-20T18:48:33.050+0000] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/dags/01_Bronce.py", line 423, in extract_and_load_table_optimized
    records_loaded = loader.load_data_streaming(data_iterator, bronze_table_name)
  File "/opt/airflow/dags/01_Bronce.py", line 336, in load_data_streaming
    for chunk in data_iterator:
  File "/opt/airflow/dags/01_Bronce.py", line 234, in extract_table_streaming
    chunk = self.optimize_dtypes(chunk)
  File "/opt/airflow/dags/01_Bronce.py", line 206, in optimize_dtypes
    if c_min > pd.np.iinfo(pd.np.int8).min and c_max < pd.np.iinfo(pd.np.int8).max:
AttributeError: module 'pandas' has no attribute 'np'
[2025-09-20T18:48:33.065+0000] {taskinstance.py:1226} INFO - Marking task as UP_FOR_RETRY. dag_id=bronze_full_load_spaceparts_optimized, task_id=extract_load_dim_Order_DocType, run_id=scheduled__2024-01-01T00:00:00+00:00, execution_date=20240101T000000, start_date=20250920T184826, end_date=20250920T184833
[2025-09-20T18:48:33.103+0000] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-09-20T18:48:33.104+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 134 for task extract_load_dim_Order_DocType (module 'pandas' has no attribute 'np'; 156)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/dags/01_Bronce.py", line 423, in extract_and_load_table_optimized
    records_loaded = loader.load_data_streaming(data_iterator, bronze_table_name)
  File "/opt/airflow/dags/01_Bronce.py", line 336, in load_data_streaming
    for chunk in data_iterator:
  File "/opt/airflow/dags/01_Bronce.py", line 234, in extract_table_streaming
    chunk = self.optimize_dtypes(chunk)
  File "/opt/airflow/dags/01_Bronce.py", line 206, in optimize_dtypes
    if c_min > pd.np.iinfo(pd.np.int8).min and c_max < pd.np.iinfo(pd.np.int8).max:
AttributeError: module 'pandas' has no attribute 'np'
[2025-09-20T18:48:33.126+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-09-20T18:48:33.155+0000] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-09-20T18:48:33.165+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
